### ISSUE TITLE
v2014.7.0+ Unstable s3fs no longer detects and uses the local cache. Always fetches files - makes it unstable to use s3fs with files even 18MB in size

### ISSUE THREAD

**Issue by hubez** [saltstack/salt#18571]
The new s3fs s3 cache detection mechanism no longer works.

In v2014.1.13, it correctly identifies that the cached file is good and doesn't need to re-download it from S3. It displays this using the follow log and runs very fast:

```
2014-11-29 02:10:47,973 [salt.loaded.int.fileserver.s3fs              ][INFO    ] mys3 - base : base/MyTest/myfile skipped download since cached file size equal to and mtime after s3 values
```

In v2014.7.0, this caching detection is broken. This log is no longer seen, and every request to the file served from s3fs is redownloaded all the time. This causes huge stability problems. Salt hangs often. Sometimes restarts of the master and minion don't recover it.

**Comment by robgott** [comment]
could dde993e14db53b2efd163ee252413d5cd725adbf possibly be responsible for this issue?
s3.query was effectively returning None for HEAD requests which is what it looks like s3fs is using to check Last-Modified and Content-Length.

PR #21551 could possibly fix the issue.

**Comment by deuscapturus** [comment]
s3fs not wokring for me on 2014.7.1 or any later version or development version.  The module repeatedly tries to retrieve the files from s3 and places them in cache, but never delivers them to the minion.  Has this ever worked?

**Comment by deuscapturus** [comment]
Workaround with cron:
2 * * * *	aws s3 sync s3://bucket-name/ /srv/salt/states/

**Comment by wkral** [comment]
I just ran into this problem. The file was uploaded with multipart upload. I ended up deleting the file and was able to upload it without multipart using the following command:

    aws s3api put-object --bucket <bucket> --key <s3 file path> --body <local file name>

Found out how to do this at https://github.com/aws/aws-cli/issues/946 it uploads in a single go and allowed salt to work with the file. This seems to be a bit of a better work around with only a few files then running a cron job to sync unless you have a large established s3 bucket already.

I used the default `aws s3 cp` command line tool initially which I feel like a lot of people will do and that caused the problem.

So...
+1
